core/commands/storage/upload/upload/dc_repair_router.go

```diff
@@ -0,0 +1,265 @@
+		for index, peerId := range peerIds {
+			go func() {
+				err = func() error {
+					pi, err := remote.FindPeer(req.Context, n, peerId)
+					if err != nil {
+						return err
+					}
+					_, err = remote.P2PCallStrings(req.Context, n, api, pi.ID, "/storage/upload/dcrepairrouting/response",
+						req.Arguments[1:]...)
+					if err != nil {
+						return err
+					}
+					reapirIds[index] = peerId
+					return nil
+				}()
+				if err != nil {
+					log.Debug("P2P call error for peerId %s", peerId)
+				}
+				wg.Done()
+			}()
+		}
+		wg.Wait()
+		return cmds.EmitOnce(res, &peerIdList{reapirIds})
+	},
+	Type: peerIdList{},
+}
+
+type peerIdList struct {
+	Strings []string
+}
+
+var hostRepairResponseCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Host responds to repair requests.",
+		ShortDescription: `
+This command enquires the repairer with the contract, if agrees with the contract after negotiation,
+returns the repairer's signed contract to the invoker.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("file-hash", true, false, "File hash for the host prepares to repair."),
+		cmds.StringArg("repair-shard-hash", true, false, "Shard hash for the host prepares to repair."),
+		cmds.StringArg("file-size", true, false, "Size of the repair file."),
+		cmds.StringArg("download-reward-amount", true, false, "Reward amount for download workload."),
+		cmds.StringArg("repair-reward-amount", true, false, "Reward amount for repair workload."),
+		cmds.StringArg("repair-pid", true, false, "Host Peer ID to send repair requests."),
+		cmds.StringArg("download_contract_id", true, false, " Contract ID associated with the download requests."),
+		cmds.StringArg("repair_contract_id", true, false, "Contract ID associated with the require requests."),
+	},
+	RunTimeout: 1 * time.Minute,
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		fileHash := req.Arguments[0]
+		repairId := req.Arguments[5]
+		downloadContractId := req.Arguments[6]
+		repairContractId := req.Arguments[7]
+		lostShardHashes := strings.Split(req.Arguments[1], ",")
+		fileSize, err := strconv.ParseInt(req.Arguments[2], 10, 64)
+		if err != nil {
+			return err
+		}
+		downloadRewardAmount, err := strconv.ParseInt(req.Arguments[3], 10, 64)
+		if err != nil {
+			return err
+		}
+		repairRewardAmount, err := strconv.ParseInt(req.Arguments[4], 10, 64)
+		if err != nil {
+			return err
+		}
+		err = emptyCheck(fileHash, lostShardHashes, repairId, repairContractId, fileSize, downloadRewardAmount, repairRewardAmount)
+		if err != nil {
+			return err
+		}
+		ctxParams, err := uh.ExtractContextParams(req, env)
+		if err != nil {
+			return err
+		}
+		ns, err := helper.GetHostStorageConfig(ctxParams.Ctx, ctxParams.N)
+		if err != nil {
+			return err
+		}
+		var price uint64
+		if ns.RepairCustomizedPricing {
+			price = ns.RepairPriceCustomized
+		} else {
+			price = ns.RepairPriceDefault
+		}
+		if price > uint64(repairRewardAmount) {
```

Comment:
amount should equal price * filesize as we should give different amount for different filesize.

Reply:
updated